AppCenterLoaderApp/appcenter-loader/src/main/java/com/microsoft/appcenter/loader/AppCenterLoader.java

```diff
@@ -45,17 +44,20 @@
     private static final String APP_SECRET_KEY = "appcenter_app_secret";
     private static final String TRANSMISSION_TARGET_TOKEN_KEY = "appcenter_transmission_target_token";
     private static final String STARTUP_TYPE_KEY = "appcenter_startup_type";
+    private static final String STARTUP_TYPE_SHARED_PREFERENCES_KEY = "AppCenter.Unity.StartTargetKey";
     private static final String TRUE_VALUE = "True";
     private static final String TAG = "AppCenterLoader";
 
     private Context mContext;
 
+    public static final String PREFS_NAME = "AppCenterUserPrefs";
+
     @Override
     public boolean onCreate() {
         mContext = getApplicationContext();
         String appSecret = getStringResource(APP_SECRET_KEY);
         String transmissionTargetToken = getStringResource(TRANSMISSION_TARGET_TOKEN_KEY);
-        int startupTypeInt = Integer.parseInt(getStringResource(STARTUP_TYPE_KEY));
+        int startupTypeInt = mContext.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE).getInt(STARTUP_TYPE_SHARED_PREFERENCES_KEY, Integer.parseInt(getStringResource(STARTUP_TYPE_KEY)));;
```

Comment:
Multiple semicolons.
Plus, it's IO on the main thread, but I'm not sure how to avoid it here